src/diet.ts

```diff
@@ -342,130 +241,317 @@ function pillCheck(): void {
+      { price: (2 * getSaleValue($item`one-day ticket to Dinseylandfill`)) / 20 }
+    ),
+
+    // BOOZE POTIONS
+    ...potion($item`dirt julep`),
+    ...potion($item`Ambitious Turkey`),
+    ...potion($item`Friendly Turkey`),
+    ...potion($item`vintage smart drink`),
+    ...limitedPotion($item`Hot Socks`, 3, { price: 5000 }),
+
+    // SPLEEN POTIONS
+    ...potion($item`beggin' cologne`),
+    ...limitedPotion($item`body spradium`, availableAmount($item`body spradium`)),
+    ...potion($item`Knob Goblin pet-buffing spray`),
+    ...potion($item`Knob Goblin nasal spray`),
+    ...potion(
+      new Potion($item`Rethinking Candy`, {
+        effect: $effect`Synthesis: Greed`,
+        duration: 30,
+      }),
+      {
+        size: 1,
+        organ: "spleen item",
+        price: 0,
+      }
+    ),
+  ];
+}
 
-  // Only use our astral pilsners if we're ascending. Otherwise they're good for shotglass.
-  if (have($item`astral pilsner`) && globalOptions.ascending) {
-    menu.push(
-      new MenuItem($item`astral pilsner`, {
-        maximum: availableAmount($item`astral pilsner`),
-      })
-    );
+interface DietPlanner {
+  (menu: MenuItem<Note>[]): Diet<Note>;
+}
+function balanceMenu(baseMenu: MenuItem<Note>[], dietPlanner: DietPlanner): MenuItem<Note>[] {
+  const baseEmbezzlers = embezzlerCount();
+  function rebalance(
+    menu: MenuItem<Note>[],
+    iterations: number,
+    embezzlers: number,
+    adventures: number
+  ): MenuItem<Note>[] {
+    const fullMenu = potionMenu(menu, baseEmbezzlers + embezzlers, estimatedTurns() + adventures);
+    if (iterations <= 0) {
+      return fullMenu;
+    } else {
+      const balancingDiet = dietPlanner(fullMenu);
+      return rebalance(
+        menu,
+        iterations - 1,
+        countCopies(balancingDiet),
+        balancingDiet.expectedAdventures()
+      );
+    }
   }
+  const baseDiet = dietPlanner(baseMenu);
+  return rebalance(baseMenu, 5, 0, baseDiet.expectedAdventures());
+}
 
+export function computeDiet(): {
+  diet: () => Diet<Note>;
+  shotglass: () => Diet<Note>;
+  pantsgiving: () => Diet<Note>;
+} {
   // Handle spleen manually, as the diet planner doesn't support synth. Only fill food and booze.
-  return planDiet(MPA, menu, [
-    ["food", null],
-    ["booze", null],
-  ]);
+
+  const fullDietPlanner = (menu: MenuItem<Note>[]) => Diet.plan(MPA, menu);
+  const shotglassDietPlanner = (menu: MenuItem<Note>[]) => Diet.plan(MPA, menu, { booze: 1 });
+  const pantsgivingDietPlanner = (menu: MenuItem<Note>[]) => Diet.plan(MPA, menu, { food: 1 });
+  // const shotglassFilter = (menuItem: MenuItem)
+
+  return {
+    diet: () => fullDietPlanner(balanceMenu(menu(), fullDietPlanner)),
+    shotglass: () =>
+      shotglassDietPlanner(
+        balanceMenu(
+          menu().filter((menuItem) => itemType(menuItem.item) === "booze" && menuItem.size === 1),
+          shotglassDietPlanner
+        )
+      ),
+    pantsgiving: () =>
+      pantsgivingDietPlanner(
```

Comment:
It feels like overkill to use the diet code for this, but I guess it probably doesn't cost us that much